### PR TITLE
Add op alias to hook

### PR DESCRIPTION
**Pull request by clairelin135**
Add aliasing to hooks, allowing users to access the following op attributes within `HookContext` (e.g. `HookContext.op`, `HookContext.op_config`, etc.)

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution/context/hook.py [hunk 1]
```diff
@@ -367,6 +386,13 @@ def build_hook_context(
             with build_hook_context(resources={"foo": context_manager_resource}) as context:
                 hook_to_invoke(context)
     """
+    if op:
+        experimental_arg_warning("op", "build_hook_context")
+        return UnboundHookContext(
+            resources=check.opt_dict_param(resources, "resources", key_type=str),
+            mode_def=ModeDefinition,
+            solid=check.opt_inst_param(op, "op", (OpDefinition, PendingNodeInvocation)),
+        )
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/execution/context/hook.py:395]: I would maybe add an invariant s.t. someone can't provide both a solid and an op, just to distinguish that they're aliases a la `check.invariant(not (solid and op))`

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Just adding that invariant and then this is probably good to go.

File: python_modules/dagster/dagster/core/execution/context/hook.py [hunk 2]
```diff
@@ -386,6 +386,7 @@ def build_hook_context(
             with build_hook_context(resources={"foo": context_manager_resource}) as context:
                 hook_to_invoke(context)
     """
+    check.invariant(not (solid and op))
```
Comment (dpeng817) [inline python_modules/dagster/dagster/core/execution/context/hook.py:389]: Add error msg + test
`check.invariant(invariant, msg)`

**Review by dpeng817 — CHANGES_REQUESTED** [review]

File: python_modules/dagster/dagster_tests/core_tests/hook_tests/test_hook_def.py [hunk 3]
```diff
@@ -465,3 +466,17 @@ def run_success_hook():
...
+    def hook_one(context):
+        pass
+
+    @op
+    def hook_op(_):
+        pass
+
+    with pytest.raises(check.CheckError):
+        ctx = build_hook_context(op=hook_op, solid=hook_op)
+        hook_one(ctx)
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/core_tests/hook_tests/test_hook_def.py:482]: Nit: I think this is dead code, I think you just need to call `build_hook_context`

**Review by dpeng817 — APPROVED** [review]
One nit, but otherwise LGTM!

File: python_modules/dagster/dagster_tests/core_tests/hook_tests/test_hook_def.py (hunk 3 above)
Comment (clairelin135) [reply python_modules/dagster/dagster_tests/core_tests/hook_tests/test_hook_def.py]: Thanks, corrected this!

**Pull request merged**